api/public/login/script.js

```diff
@@ -25,7 +25,9 @@ var post = function(url, payload, callback) {
 };
 
 var handleResponse = function(xmlhttp) {
-  if (xmlhttp.status === 200) {
+  if (xmlhttp.readyState === XMLHttpRequest.DONE && xmlhttp.status < 200) {
```

Comment:
Is checking the readyState necessary? It _should_ be checked by line 18 so this function isn't called until readyState === DONE. If not, that's a bug we need to fix.

Reply:
Agree. Good catch.